ic-asset/src/asset_config.rs

```diff
@@ -0,0 +1,627 @@
+use anyhow::bail;
+use derivative::Derivative;
+use globset::{Glob, GlobMatcher};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{
+    cell::{RefCell, RefMut},
+    collections::HashMap,
+    fs::File,
+    io::BufReader,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+const ASSETS_CONFIG_FILENAME: &str = ".ic-assets.json";
+
+thread_local! {
+    // There is no simple way to pass state into serde deserializer. Using this as a walkaround.
+    // https://stackoverflow.com/questions/54052495/mock-instance-inside-serde-implementation
+    static CURRENTLY_PROCESSED_ASSETS_DIRECTORY: RefCell<PathBuf> = RefCell::new(PathBuf::new());
+}
+
+pub(crate) type HeadersConfig = HashMap<String, Value>;
+type Map = HashMap<PathBuf, Arc<AssetConfigTreeNode>>;
+
+#[derive(Deserialize, Serialize, Debug, Default, Clone, PartialEq, Eq)]
+pub(crate) struct CacheConfig {
+    pub(crate) max_age: u64,
```

Comment:
To make it easier to add fields to this struct in the future:
```suggestion
    pub(crate) max_age: Option<u64>,
```

Reply:
done in https://github.com/dfinity/agent-rs/pull/352/commits/5cd6a6db74324d34ba6381b550905b2597f1eb71